### DISCUSSION TITLE
Creating non-circular holes in sketches

### DISCUSSION THREAD

**Discussion by raydeleu** [zalo/CascadeStudio#94] [Q&A]
![honeycomb_large](https://user-images.githubusercontent.com/38007983/118498681-96b4be80-b726-11eb-828e-187ddda0c551.png)

I was trying to create a honeycomb in CascadeStudio, using several approaches. First I tried to subtract hexagonal columns from a target shape. My second attempt involved creating a zig-zag sketch, then extruding, mirroring and translating these into a full honeycomb, creating a new Union on every iteration. Both approaches however take a considerable amount of processing time as the size of the honeycomb increases. 

Is there a way to create a sketch with hexagonal holes and extrude this in one go? I only found and succesfully tested this with circular holes, but I found no way to create other shapes (apart from creating somthing like a closed wall around a single hole). I tried sketch.Start(new starting point) but this seems to erase the earlier sketch.

**Comment by zalo** [comment]
You could try using the Pipe function to extrude a vertical rectangle along a zig-zag path, but that's it for other ways I can think of doing this...
